### ISSUE TITLE
Support for codesigning binaries

### ISSUE THREAD

**Issue by conradev** [facebook/xcbuild#44]
The ability to codesign binaries upon build would be great to have, reading from `CODE_SIGN_IDENTITY` and `CODE_SIGN_ENTITLEMENTS`.

One thing I didn't realize until recently is that [ldid](http://gitweb.saurik.com/ldid.git) supports full codesigning, so to be able to use either `codesign` (OS X-only) or `ldid` (cross platform) to sign the binary would be awesome.

**Comment by samdmarshall** [comment]
the codesign binaries are fully able to be compiled from source. I don't see why we cannot use those.

**Comment by conradev** [comment]
@samdmarshall Do they have OS X-specific dependencies?

**Comment by samdmarshall** [comment]
I would have to double check, but I don't think there are any additional requirements that wouldn't also be able to be compiled from source.

**Comment by kphillisjr** [comment]
From what I can figure out codesigning on OSX is mostly a post-process development phase. Details on this can be found at these two links:

[Apple: Mac Developer Library: Code Signing Guide.](https://developer.apple.com/library/mac/documentation/Security/Conceptual/CodeSigningGuide/Introduction/Introduction.html)

[Apple: Mac Developer Library: Technical Note TN2206: OS X Code Signing In Depth](https://developer.apple.com/library/mac/technotes/tn2206/_index.html)

**Comment by grp** [comment]
Yeah, it is. It's fairly simple to sign binaries yourself with `codesign -s <identity> /path/to.app` after a build, but it would be better to integrate into the build as it should be.
